File: spec/javascript/test_enum_if.js
var assert = require('assert');
var testHelper = require('testHelper');

testHelper('EnumIf', 'src/if_struct.bin', function(r, EnumIf) {
    assert.equal(r.op1.opcode, EnumIf.Opcodes.A_STRING);
    assert.equal(r.op1.argStr.str, "foo");

    assert.equal(r.op2.opcode, EnumIf.Opcodes.A_TUPLE);
    assert.equal(r.op2.argTuple.num1, 0x42);
    assert.equal(r.op2.argTuple.num2, 0x43);

    assert.equal(r.op3.opcode, EnumIf.Opcodes.A_STRING);
    assert.equal(r.op3.argStr.str, "bar");
});

File: spec/lua/test_enum_if.lua
local luaunit = require("luaunit")

require("enum_if")

TestEnumIf = {}

function TestEnumIf:test_enum_if()
    local r = EnumIf:from_file("src/if_struct.bin")

    luaunit.assertEquals(r.op1.opcode, EnumIf.Opcodes.a_string)
    luaunit.assertEquals(r.op1.arg_str.str, "foo")

    luaunit.assertEquals(r.op2.opcode, EnumIf.Opcodes.a_tuple)
    luaunit.assertEquals(r.op2.arg_tuple.num1, 0x42)
    luaunit.assertEquals(r.op2.arg_tuple.num2, 0x43)

    luaunit.assertEquals(r.op3.opcode, EnumIf.Opcodes.a_string)
    luaunit.assertEquals(r.op3.arg_str.str, "bar")
end
